From f324c122629ee07c7fef56496b8efb331fdaf0c8 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Sun, 6 Feb 2022 01:29:31 -0600
Subject: [PATCH 1/2] [BUG] Fix importing scipy.lib._pep440

- Change scipy/_lib/_pep440 to scipy._lib._pep440
- Do the second import in else clause to avoid errors like this
  in the future
- Set the imported module to a variable
---
 setup.py | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/setup.py b/setup.py
index a5ccaa159..e7ba1c45a 100755
--- a/setup.py
+++ b/setup.py
@@ -232,16 +232,17 @@ def get_build_ext_override():
         try:
             import pythran
             from pythran.dist import PythranBuildExt
+        except ImportError:
+            BaseBuildExt = npy_build_ext
+        else:
             BaseBuildExt = PythranBuildExt[npy_build_ext]
-            importlib.import_module('scipy/_lib/_pep440')
+            _pep440 = importlib.import_module('scipy._lib._pep440')
             if _pep440.parse(pythran.__version__) < _pep440.Version('0.10.0'):
                 raise RuntimeError("The installed `pythran` is too old, >= "
                                    "0.10.0 is needed, {} detected. Please "
                                    "upgrade Pythran, or use `export "
                                    "SCIPY_USE_PYTHRAN=0`.".format(
                                    pythran.__version__))
-        except ImportError:
-            BaseBuildExt = npy_build_ext
     else:
         BaseBuildExt = npy_build_ext
 
@@ -345,7 +346,10 @@ def generate_cython():
         try:
             # Note, pip may not be installed or not have been used
             import pip
-            importlib.import_module('scipy/_lib/_pep440')
+        except (ImportError, ModuleNotFoundError):
+            raise RuntimeError("Running cythonize failed!")
+        else:
+            _pep440 = importlib.import_module('scipy._lib._pep440')
             if _pep440.parse(pip.__version__) < _pep440.Version('18.0.0'):
                 raise RuntimeError("Cython not found or too old. Possibly due "
                                    "to `pip` being too old, found version {}, "
@@ -353,8 +357,6 @@ def generate_cython():
                                    pip.__version__))
             else:
                 raise RuntimeError("Running cythonize failed!")
-        except (ImportError, ModuleNotFoundError):
-            raise RuntimeError("Running cythonize failed!")
 
 
 def parse_setuppy_commands():
-- 
2.32.0.windows.2

